python/ray/train/base_trainer.py

```diff
@@ -379,6 +370,14 @@ class TrainTrainable(trainable_cls):
             def __repr__(self):
                 return super().__repr__()
 
+            @classmethod
+            def has_dataset(cls):
+                return has_dataset
+
+            @classmethod
+            def base_scaling_config(cls):
+                return scaling_config
```

Comment:
Add a comment explaining that these methods only contain the original dataset/scaling config that was used when instantiating the Trainer, but are not indicative of updates when tuning over these as hyperparameters?

I also think we should rename to `has_base_dataset` to be more explicit